MapzenSDK/MZMapViewController.swift

```diff
@@ -487,8 +487,10 @@ open class MZMapViewController: UIViewController, LocationManagerDelegate {
    - parameter marker: The marker to add to map.
    */
   open func addMarker(_ marker: GenericMarker) {
-    currentMarkers[marker.tgMarker] = marker
-    marker.tgMarker.map = tgViewController
+    if marker.tgMarker == nil {
+      marker.tgMarker = tgViewController.markerAdd()
+    }
+    currentMarkers[marker.tgMarker!] = marker
```

Comment:
Ok so @sarahlensing and I discussed it further and I think we wanna go down the route of using the user data object on TGMarker. However, a void* there imported into Swift is difficult to use. Do you think it would be much effort to convert it to a nullable `NSObject` ?